src/Symfony/Component/Form/Tests/Extension/Core/Type/DateTypeTest.php

```diff
@@ -358,10 +382,25 @@ public function testThrowExceptionIfDaysIsInvalid()
 
     public function testSetDataWithDifferentNegativeUTCTimezoneDateTime()
     {
-        date_default_timezone_set('Pacific/Tahiti');
+        $form = $this->factory->create('date', null, array(
+            'format' => \IntlDateFormatter::MEDIUM,
+            'model_timezone' => 'America/New_York',
+            'view_timezone' => 'Pacific/Tahiti',
+            'input' => 'string',
+            'widget' => 'single_text',
+        ));
 
+        $form->setData('2010-06-02');
+
+        $this->assertEquals('01.06.2010', $form->getViewData());
```

Comment:
This particular test shows this two options are pointless and might be misleading in case of Date/Birthday/Time types. It doesn't matter in which timezone a person was born, changing a timezone shouldn't change a date of birth.